src/validation.cpp

```diff
@@ -2095,17 +2095,15 @@ bool CScriptCheck::operator()() {
     return VerifyScript(scriptSig, m_tx_out.scriptPubKey, witness, nFlags, CachingTransactionSignatureChecker(ptxTo, nIn, m_tx_out.nValue, cacheStore, *txdata), &error);
 }
 
-static CSHA256 g_scriptExecutionCacheHasher;
-
 ValidationCache::ValidationCache(size_t script_execution_cache_bytes)
 {
     // Setup the salted hasher
     uint256 nonce = GetRandHash();
     // We want the nonce to be 64 bytes long to force the hasher to process
     // this chunk, which makes later hash computations more efficient. We
     // just write our 32-byte entropy twice to fill the 64 bytes.
-    g_scriptExecutionCacheHasher.Write(nonce.begin(), 32);
-    g_scriptExecutionCacheHasher.Write(nonce.begin(), 32);
+    m_script_execution_cache_hasher.Write(nonce.begin(), 32);
```

Comment:
Might be useful to pass the nonce in rather than generating it here to make the order of rng/ValidationCache instantiation explicit.

Presumably it'd also make for easier testing with a deterministic seed.

Reply:
Does this even need to be a fresh random value every time? Couldn't it just be any nothing up my sleeve number?

Reply:
I believe the idea behind these nonces is to introduce diversity so that everyone has a different cache. That way there's no possibility of poisoning the "global" cache.

Reply:
I did this now, but was not completely sure where to put the nonce generation.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/30141#discussion_r1658934242

> I did this now, but was not completely sure where to put the nonce generation.

For reference, this was implemented in 45e0c96e81b5f0c364af717b7cb2d9bc094372de but later dropped from the PR